sdk/digitaltwins/digitaltwins/samples/javascript/dt_component_lifecycle.js

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+const { DefaultAzureCredential } = require("@azure/identity");
+const { DigitalTwinsClient } = require("@azure/digitaltwins");
+const { v4 } = require("uuid");
+
+// Scenario example of how to:
+// - create a DigitalTwins Service Client using the DigitalTwinsClient constructor
+// - create model, component and twin
+// - create digital twin based on the model
+// - update component
+// - get component
+// - delete twin
+// - decomission and delete model, component
+//
+// Preconditions:
+// - Environment variables have to be set
+// - DigitalTwins enabled device must exist on the ADT hub
+//
+// For the purpose of this example we will create temporary digital twin using random Ids.
+// We have to make sure these model Ids are unique within the DT instance so we use generated UUIDs.
+async function main() {
+  const modelId = "model-" + v4();
+  const componentId = "component-" + v4();
+  const digitalTwinId = "digitalTwin-" + v4();
+
+  const temporaryComponent = {
+    "@id": componentId,
+    "@type": "Interface",
+    "@context": "dtmi:dtdl:context;2",
+    displayName: "Component1",
+    contents: [
+      {
+        "@type": "Property",
+        name: "ComponentProp1",
+        schema: "string"
+      }
+    ]
+  };
+
+  const temporaryModel = {
+    "@id": modelId,
+    "@type": "Interface",
+    "@context": "dtmi:dtdl:context;2",
+    displayName: "TempModel",
+    contents: [
+      {
+        "@type": "Property",
+        name: "Prop1",
+        schema: "double"
+      },
+      {
+        "@type": "Component",
+        name: "Component1",
+        schema: componentId
+      }
+    ]
+  };
+
+  const temporaryTwin = {
+    "@id": digitalTwinId,
+    $metadata: {
+      "@model": modelId
+    },
+    Prop1: 42,
+    Component1: {
+      $metadata: {},
+      ComponentProp1: "value1"
+    }
+  };
+
+  // - AZURE_URL: The tenant ID in Azure Active Directory
+  const url = process.env.AZURE_URL;
```

Comment:
is this SDK Node only? We should call that out in the README, since many of our client packages work in the browser

Reply:
No it supposed to run in browser also.

---
In reply to: [447258241](https://github.com/Azure/azure-sdk-for-js/pull/9787#discussion_r447258241) [](ancestors = 447258241)